src/fparser/common/readfortran.py

```diff
@@ -1221,21 +1273,35 @@ def get_source_item(self):
         A source item is ..
         - a fortran line
         - a list of continued fortran lines
-        - a multiline - lines inside triple-qoutes, only when in ispyf mode
+        - a multiline - lines inside triple-quotes, only when in ispyf mode
         - a comment line
+        - a preprocessor directive line
         """
         get_single_line = self.get_single_line
         line = get_single_line()
         if line is None:
             return
         startlineno = self.linecount
+        line, is_cpp_directive = self.handle_cpp_directive(line)
         line = self.handle_cf2py_start(line)
         is_f2py_directive = startlineno in self.f2py_comment_lines
         isstrict = self._format.is_strict
         have_comment = False
         label = None
         name = None
 
+        if is_cpp_directive:
```

Comment:
Out of interest, why do you separate this additional code from the additional code on line 1285. Do the lines in-between need to be done for the code to work? I can't see why. Sorry if it is obvious.

Reply:
Good point. I'll move ll. 1293ff. up.